### PR TITLE
Live recovery

### PR DESCRIPTION
**Pull request by jfckm**
This change is [Reviewable](https://reviewable.io/reviews/open-cas/spdk/7)

### CODE REVIEW

File: module/bdev/ocf/ctx.c [hunk 1]
```diff
@@ -533,6 +539,195 @@ vbdev_ocf_ctx_log_printf(ocf_logger_t logger, ocf_logger_lvl_t lvl,
...
+		goto err_stat;
+	}
+
+	pmeta->data = shm;
+	shm_sb = shm;
+
+	if (mlock(shm, pmeta->size)) {
+		SPDK_ERRLOG("Failed to mlock\n");
+
+		goto err_mlock;
```
Comment (mmkayPL) [inline module/bdev/ocf/ctx.c:704]: If the locking failed, shouldn't this skip executing munlock?

File: module/bdev/ocf/ctx.c [hunk 2]
```diff
@@ -533,6 +539,195 @@ vbdev_ocf_ctx_log_printf(ocf_logger_t logger, ocf_logger_lvl_t lvl,
...
+	shm_unlink(pmeta->name);
+
+	return NULL;
+}
+
+static int
+vbdev_ocf_persistent_meta_deinit(struct ocf_persistent_meta_zone *zone)
+{
+	/*
+	munmap(zone->data, zone->size);
```
Comment (mmkayPL) [inline module/bdev/ocf/ctx.c:730]: Does munlock need to be called here as well?

File: module/bdev/ocf/ctx.c [hunk 3]
```diff
@@ -533,6 +539,195 @@ vbdev_ocf_ctx_log_printf(ocf_logger_t logger, ocf_logger_lvl_t lvl,
...
+	desc[i].size = size;
+	desc[i].valid = true;
+
+	return zone->data + desc[i].offset;
+}
+
+static int
+vbdev_ocf_persistent_meta_free(struct ocf_persistent_meta_zone *zone, int alloc_id,
+		void *ptr)
+{
```
Comment (mmkayPL) [inline module/bdev/ocf/ctx.c:784]: Should this invalidate the appropriate ocf_cache_persistent_meta_segment_descriptor ?

File: module/bdev/ocf/vbdev_ocf.c [hunk 4]
```diff
@@ -1619,6 +1624,9 @@ vbdev_ocf_examine_disk(struct spdk_bdev *bdev)
 	const char *bdev_name = spdk_bdev_get_name(bdev);
 	struct vbdev_ocf *vbdev;
 
+	if (getenv("ocf_force"))
```
Comment (mmkayPL) [inline module/bdev/ocf/vbdev_ocf.c]: Is this needed? Is it different from --force flag?

Comment (jfckm) [reply module/bdev/ocf/vbdev_ocf.c]: fixed

File: module/bdev/ocf/ctx.c (hunk 3 above)
Comment (jfckm) [reply module/bdev/ocf/ctx.c:784]: this is intentionally left blank for now

File: module/bdev/ocf/ctx.c (hunk 2 above)
Comment (jfckm) [reply module/bdev/ocf/ctx.c:730]: munlock isn't needed before munmap

File: module/bdev/ocf/ctx.c (hunk 1 above)
Comment (jfckm) [reply module/bdev/ocf/ctx.c:704]: fixed

**Review by mmkayPL — DISMISSED** [review]
> Reviewed 1 of 4 files at r1, 4 of 5 files at r6.
> *[Reviewable](https://reviewable.io/reviews/open-cas/spdk/7)* status: 4 of 5 files reviewed, 5 unresolved discussions (waiting on @jfckm and @mmkayPL)

___
*[dpdk, line 1 at r6](https://reviewable.io/reviews/open-cas/spdk/7#-M_eqm0G0tBI1QVL7gyW:-M_eqm0H1Rjy4DFNNmgY:b-u2r76) ([raw file](https://github.com/open-cas/spdk/blob/77bee4c0fe8e1fd34f676bd5568c91359e41f6e1/dpdk#L1)):*
> ```
> Subproject commit 707692e67d0c6c685c42f5dd48f8f112bde6b381
> ```

Is this update intended?

**Review by mmkayPL — APPROVED** [review]

**Review by arutk — APPROVED** [review]

**Pull request merged**